packages/client/src/lib.js

```diff
@@ -155,6 +159,75 @@ class FilecoinStorage {
   }
 }
 
+/**
+ * Upgrade a ReadableStream to an AsyncIterable if it itn't already
+ *
+ * ReadableStream (e.g res.body) is asyncIterable in node, but not in chrome, yet.
+ * see: https://bugs.chromium.org/p/chromium/issues/detail?id=929585
+ *
+ * @param {ReadableStream<Uint8Array> | ReadableStream<Uint8Array> & AsyncIterable<Uint8Array>} readable
+ * @returns {AsyncIterable<Uint8Array>}
+ */
+function asAsyncIterable(readable) {
+  // @ts-ignore how to tell tsc that we are checking the type here?
+  return Symbol.asyncIterator in readable
+    ? readable
+    : /* c8 ignore next */
+      toIterable(readable)
+}
+
+/**
+ * map a UnixFSEntry to a ~Blob~ File with benefits
+ * @param {import('./lib/interface.js').UnixFSEntry} e
+ * @returns {Promise<import('./lib/interface.js').IpfsFile>}
+ */
+async function toIpfsFile(e) {
+  const chunks = []
+  for await (const chunk of e.content()) {
+    chunks.push(chunk)
+  }
+
+  // A Blob in File clothing
+  const file = Object.assign(new Blob(chunks), {
```

Comment:
@vasco-santos @alanshaw To make a browser `File` like thing, I am consuming the UnixFSEntry content... and i'm doing this for both the `res.file` method and the `res.filesIterator` which means, for big big files, you are probably going to have a bad, memory constrained time.

Should `filesIterator` be `unixFsEntryIterator` or similar... with that, we preserve all the information like directory nodes, and in the cli i could just import the existing `writeFiles` function from `ipfs-car` that takes the `AsyncIterable<UnixFsEntry>` and figures out how to write it to disk. 
https://github.com/vasco-santos/ipfs-car/blob/6935c9000703f7d2b0df529688c983987db6fce8/src/unpack/fs.ts#L29-L52

Reply:
Sounds sensible to me.

Reply:
It can, but I would make it a `ReadableFile` per https://cryptpad.fr/code/#/2/code/edit/lLebfkgGeoMJmkkuhPQS8rTN/
which is really a subset of UnixFsEntry, but something more simple that a user can consume and we can create utilities to help consuming the get content.

I would see us doing something like

```js
const putFiles = readableFiles.fromFs(path: String)
const cid = await put (putFiles: Iterable<ReadableFile>)
const files = await get(cid: CID)
await readableFiles.toFs(files) // Back in fs
```

While for browser users just need to provide a ReadableFile Iterator (Blob, String, Uint8Array) + get one, which should be easy to parse. Using WebFile, we can also provide a `readableFiles.FromWebFile()`, which would convert Web File to Readable File, if users use the picker.

Reply:
`readableFiles.fromFs` would go through the provided path and create an iterable of

```
{ path: String, content: ReadableStream }
```

which ipfsCar pack would accept.